src/nvim/ex_getln.c

```diff
@@ -1342,6 +1344,7 @@ static int command_line_handle_key(CommandLineState *s)
 
   case Ctrl_R:                        // insert register
     putcmdline('"', true);
+    extra_char = '"';
```

Comment:
We already have `ccline.special_char` in nvim for this, but it needs to be made unconditional in `putcmdline`